### PR TITLE
Prevent show spinner when drag Leo panel on Android

### PR DESCRIPTION
**Pull request by AlexeyBarabash**
Fixes brave/brave-browser#35838

Resolves https://github.com/brave/brave-browser/issues/35838

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

#### I. Normal flow with Leo panel
See the issue.

####  II. Ensure default behavior is the same for non-Leo cases
1. Make Brave Browser default
2. Launch and exit Brave Browser - just in case - otherwise on my device it kept opening Chrome in custom tab somehow. 
3. Install https://github.com/GoogleChrome/android-browser-helper/tree/main/demos/custom-tabs-example-app . You can build it by yourslef or take the one I've built 
[custom-tabs-example-app-debug.zip](https://github.com/brave/brave-core/files/14192710/custom-tabs-example-app-debug.zip)
4. Launch `Custom Tabs Demos`
5. Press `Partial Custom Tab` section
6. Press `OPEN CUSTOM TAB` button
7. Drag tab panel to the top, ensure you see spinner

### CODE REVIEW

**Review by samartnik — APPROVED** [review]
lgtm

File: android/java/org/chromium/chrome/browser/customtabs/features/partialcustomtab/BravePartialCustomTabBottomSheetStrategy.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                startMaximized,
+                handleStrategyFactory);
+    }
+
+    public boolean mStopShowingSpinner;
+
+    @Override
+    public void onDragStart(int y) {
+        super.onDragStart(y);
+        mStopShowingSpinner = true;
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/customtabs/features/partialcustomtab/BravePartialCustomTabBottomSheetStrategy.java:50]: let's do that for our tab only, just make a detection like you did for a page refresh for the schema and chat URL. We better don't break other pages inside custom tabs.

Comment (AlexeyBarabash) [reply android/java/org/chromium/chrome/browser/customtabs/features/partialcustomtab/BravePartialCustomTabBottomSheetStrategy.java]: Sorry, missed this point though we had discussed.
Fixed at https://github.com/brave/brave-core/pull/21927/commits/ba776f2dacb6a6a601c5f5a48508ae16c9557718 .

**Review by SergeyZhukovsky — APPROVED** [review]
++

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Pixel 8` running `Android 14` using the following build(s):

```
Brave | 1.64.65 Chromium: 122.0.6261.29 (Official Build) canary (64-bit)
--- | ---
Revision | a31a4290998445a19f5f22b803256e97ec3036a5
OS | Android 14; Build/UQ1A.240105.004; 34; REL
```

Using the STR/Cases outlined via https://github.com/brave/brave-browser/issues/35838#issue-2121435197, ensured that the refresh spinner isn't visible when dragging `Leo` as per the following:

https://github.com/brave/brave-core/assets/2602313/6ff9b0d0-9924-4b28-a4a2-8ed82b448be1

**Pull request merged**